pkg/controller/qdrouterd/qdrouterd_controller.go

```diff
@@ -231,6 +251,97 @@ func (r *ReconcileQdrouterd) Reconcile(request reconcile.Request) (reconcile.Res
+			controllerutil.SetControllerReference(instance, newIssuer, r.scheme)
+			reqLogger.Info("Creating a new self signed issuer %s%s\n", newIssuer.Namespace, newIssuer.Name)
+			err = r.client.Create(context.TODO(), newIssuer)
+			if err != nil {
+				reqLogger.Info("Failed to create new self signed issuer", "error", err)
+				return reconcile.Result{}, err
+			}
+			// Issuer created successfully - return and requeue
+			return reconcile.Result{Requeue: true}, nil
+		} else if err != nil {
+			reqLogger.Info("Failed to get self signed issuer", "error", err)
+			return reconcile.Result{}, err
+		}
+	}
+
+	if requestCert && instance.Spec.Issuer == "" {
+		selfSignedCertFound := &cmv1alpha1.Certificate{}
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: instance.Name + "-selfsigned", Namespace: instance.Namespace}, selfSignedCertFound)
+		if err != nil && errors.IsNotFound(err) {
+			// Create a new self signed certificate
+			cert := certificates.NewSelfSignedCertificateForCR(instance)
+			controllerutil.SetControllerReference(instance, cert, r.scheme)
+			reqLogger.Info("Creating a new self signed cert %s%s\n", cert.Namespace, cert.Name)
+			err = r.client.Create(context.TODO(), cert)
+			if err != nil {
+				reqLogger.Info("Failed to create new self signed cert", "error", err)
+				return reconcile.Result{}, err
+			}
+			// Cert created successfully - return and requeue
+			return reconcile.Result{Requeue: true}, nil
+		} else if err != nil {
+			reqLogger.Info("Failed to create self signed cert", "error", err)
+			return reconcile.Result{}, err
+		}
+		caSecretName = selfSignedCertFound.Name
+	}
+
+    // If needed, check if CA issuer exists and if not create a new one
+	if requestCert {
+		caIssuerFound := &cmv1alpha1.Issuer{}
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: instance.Name + "-ca", Namespace: instance.Namespace}, caIssuerFound)
+		if err != nil && errors.IsNotFound(err) {
+			// Define a new ca issuer
+			newIssuer := certificates.NewCAIssuerForCR(instance, caSecretName)
+			controllerutil.SetControllerReference(instance, newIssuer, r.scheme)
+			reqLogger.Info("Creating a new ca issuer %s%s\n", newIssuer.Namespace, newIssuer.Name)
+			err = r.client.Create(context.TODO(), newIssuer)
+			if err != nil {
+				reqLogger.Info("Failed to create new ca issuer", "error", err)
+				return reconcile.Result{}, err
+			}
+			// Issuer created successfully - return and requeue
+			return reconcile.Result{Requeue: true}, nil
+		} else if err != nil {
+			reqLogger.Info("Failed to get ca issuer", "error", err)
+			return reconcile.Result{}, err
+		}
+	}
+
+	// If needed, create certs for SslProfiles
+	if requestCert {
+		for _, profile := range instance.Spec.SslProfiles {
+			caCertFound := &cmv1alpha1.Certificate{}
+			err = r.client.Get(context.TODO(), types.NamespacedName{Name: instance.Name + "-" + profile.Name, Namespace: instance.Namespace}, caCertFound)
```

Comment:
I think we only want to create certificates where the user did not define the necessary secrets. They may not use cert manager for doing so, so testing for a certificate cr is I think the wrong check.

My thought was to check if profile.Credentials == "" and if so create a cert based on the ssl profile name (and router instance name). Likewise if profile.RequestClientCerts && profile.CaCert == "" then create a CA cert. The templating logic that generates the config could then assume a particular secret name if these were not specified.